cmd/nitro/nitro.go

```diff
@@ -934,3 +975,82 @@ func testUpdateTxIndex(chainDb ethdb.Database, chainConfig *params.ChainConfig)
 	}
 	log.Info("Tx lookup entries written")
 }
+
+type LiveNodeConfig struct {
+	mutex  sync.RWMutex
+	args   []string
+	config *NodeConfig
+}
+
+func (c *LiveNodeConfig) get() *NodeConfig {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+	return c.config
+}
+
+func (c *LiveNodeConfig) set(config *NodeConfig) error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if err := c.config.CanReload(config); err != nil {
+		return err
+	}
+	if err := initLog(config.LogType, log.Lvl(config.LogLevel)); err != nil {
+		return err
+	}
+	c.config = config
+	return nil
+}
+
+func (c *LiveNodeConfig) Start(ctx context.Context) {
+	sigusr1 := make(chan os.Signal, 1)
+	signal.Notify(sigusr1, syscall.SIGUSR1)
+
+	go func() {
+		for {
+			c.mutex.RLock()
```

Comment:
if we can assume that LiveNodeConfig.set is not called anywhere else then here in Start and that Start is called only once, then we can skip locking read lock. The assumption should be true in production, but is not in test (skipping the rlock causes race condition in tests)